src/adapters/openai/helpers/embedding.ts

```diff
@@ -0,0 +1,25 @@
+import OpenAI from "openai";
+import { Context } from "../../../types";
+import { SuperOpenAi } from "./openai";
+const VECTOR_SIZE = 512;
+
+export class Embedding extends SuperOpenAi {
+  protected context: Context;
+
+  constructor(client: OpenAI, context: Context) {
+    super(client, context);
+    this.context = context;
+  }
+
+  async createEmbedding(text: string): Promise<number[]> {
+    const params: OpenAI.EmbeddingCreateParams = {
+      model: "text-embedding-3-small",
```

Comment:
Is this the best model?

Reply:
<img width="862" alt="image" src="https://github.com/user-attachments/assets/980a4812-5566-4c2a-9689-d77602ec0d4c">

Not the best model, but gets the job done. Should be good enough for smaller length messages and lower dim vectors.

Reply:
- Vector search with large model could be slow with large number of entries.
- The larger dimensions size could be useful if we are designing a RAG based system, but for other purposes a smaller dimension size of around 512 to 1024 should be enough.
- The data chunks are smaller, so we could get away with a smaller vector dim.

Reply:
I think we should optimize as needed instead of proactively. This is because we are just assuming that performance may some day be a problem. 

Right now I consider this more of a research project, and honestly have basically no experience building these sort of systems.

Reply:
Just to be clear, this is a design issue as well. Depending on the type of model we choose we need to change the datatype in the vector store. Increasing the vector embedding dimension, after the vector store has been populated would be a little difficult.

And, I do agree with what you're saying, but performance and cost aside, these models perform the more or less the same on MTEB benchmark as well. I usually try to start with smaller models, and work my way up, if the application requires it.

Reply:
Fixed in the latest commit, uses large model.